modules/scheduler/gke-cluster/main.tf

```diff
@@ -30,6 +30,12 @@ locals {
   }]
 
   sa_email = var.service_account_email != null ? var.service_account_email : data.google_compute_default_service_account.default_sa.email
+
+  # additional VPCs enable multi networking 
+  derived_enable_multi_networking = length(var.additional_networks) > 0 ? true : var.enable_multi_networking
+
+  # multi networking needs enabled Dataplane v2
+  derived_enable_dataplane_v2 = local.derived_enable_multi_networking ? true : var.enable_dataplane_v2
```

Comment:
Here if user explicitly sets `enable_dataplane_v2` to false we silently ignore their setting. I do not like to do this.

I think we should do the following:
- change var.enable_dataplane_v2 to a string and use it like an enum
- have default be something like "SELECT_DEFAULT"
- provide explicit options for "ENABLED" and "DISABLED"
- if "SELECT_DEFUALT" is selected then we can override
- use precondition to ensure that we don't have conflicting options

Reply:
Great point, should we also add a validation somewhere that multi-networking requires `Dataplane_v2` so that it is not a silent failure.

Reply:
@ankitkinra: yes.
my last bullet: `use precondition to ensure that we don't have conflicting options`

A pre-condition is the Terraform way of performing validation when you need access to more than 1 variable.

Reply:
I kept it boolean but set null as default value. I also added a validation in the variable so that we prevent when enable_dataplane_v2 is explicitly set to false and enable_multinetworking is set to true.

Reply:
I think we need to add a validation to ensure enable_multi_networking is not explicitly set to false when a multivpc is passed to the gke-cluster.

Reply:
Validations replaced with preconditions. 
enable_multi_networking also was checked to prevent explicitly set "false" value when a multivpc is passed.